### ISSUE TITLE
[BUG] v7.0.0 Elixir Deserialization Regression

### ISSUE THREAD

**Issue by tobbbles** [OpenAPITools/openapi-generator#16412]
#### Bug Report Checklist

- [X] Have you provided a full/minimal spec to reproduce the issue?
- [X] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [X] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [X] Have you searched for related issues/PRs?
- [X] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

#16061 appears to have introduced a bug when attempting to deserialize a string into an enum model.

```
    ** (FunctionClauseError) no function clause matching in Ory.Deserializer.to_struct/2
        (ory_client 1.1.50) lib/ory/deserializer.ex:97: Ory.Deserializer.to_struct("aal1", Ory.Model.AuthenticatorAssuranceLevel)
```
Here we see `to_struct/2` being called on the value of "aal1", to be decoded into the `Ory.Model.AuthenticatorAssuranceLevel` model. However there is no implementation from this.

This is a regression from the prior generators, and the expected output is to be able to deserialize into these modules/structs.

##### openapi-generator version

v7.0.0

##### OpenAPI declaration file content or url

Any string-typed enum field, for example this from the [Ory SDK](https://github.com/ory/sdk/blob/0f367b584c4a332592c33653d58580f1a3ed615c/spec/client/v1.1.51.json) 
```json
      "authenticatorAssuranceLevel": {
        "description": "The authenticator assurance level can be one of \"aal1\", \"aal2\", or \"aal3\". A higher number means that it is harder\nfor an attacker to compromise the account.\n\nGenerally, \"aal1\" implies that one authentication factor was used while AAL2 implies that two factors (e.g.\npassword + TOTP) have been used.\n\nTo learn more about these levels please head over to: https://www.ory.sh/kratos/docs/concepts/credentials",
        "enum": [
          "aal0",
          "aal1",
          "aal2",
          "aal3"
        ],
        "title": "Authenticator Assurance Level (AAL)",
        "type": "string"
      },


```

##### Generation Details
The above spec will  output the following model module.

```elixir
# NOTE: This file is auto generated by OpenAPI Generator 7.0.0 (https://openapi-generator.tech).
# Do not edit this file manually.

defmodule Ory.Model.AuthenticatorAssuranceLevel do
  @moduledoc """
  The authenticator assurance level can be one of \"aal1\", \"aal2\", or \"aal3\". A higher number means that it is harder for an attacker to compromise the account.  Generally, \"aal1\" implies that one authentication factor was used while AAL2 implies that two factors (e.g. password + TOTP) have been used.  To learn more about these levels please head over to: https://www.ory.sh/kratos/docs/concepts/credentials
  """

  @derive Jason.Encoder
  defstruct [
    
  ]

  @type t :: %__MODULE__{
    
  }

  def decode(value) do
    value
  end
end


```

##### Steps to reproduce
Generate any OpenAPI schema with an enum string model and you will get this. For further testing using the Ory SDK, you can checkout [my fork](https://github.com/tobbbles/sdk) and generate the elixir client by commenting out  lines 235-333 in `scripts/generate.sh` and running `$ FORCE_PROJECT=client FORCE_VERSION=(cat spec/client/latest) ./scripts/generate.sh` to invoke the openapi-generator-cli to generate the client at `clients/client/elixir/`.

##### Related issues/PRs

#16061 Introduced this regression 

##### Suggest a fix.

Implement a way to deserialize enum modules. Ideally this would go further and implement a word list to validate whether a string value is a valid enum when trying to deserialize into the

**Comment by tobbbles** [comment]
This is still broken on the new implementation and blocking usage of openapi-generator in Elixir for any string enum components.

**Comment by barttenbrinke** [comment]
@tobbbles Sorry, this only just appeared in my inbox for some reason /cc @wpiekutowski

**Comment by barttenbrinke** [comment]
TLDR - Swagger and thus the openapi generator sees the Enum as a authenticatorAssuranceLevel struct, while it basically is just a string. As it is not really doable to handle enums in a struct like this in Elixir, just returning the string might just be the simplest solution here.

**Comment by tobbbles** [comment]
Hi @barttenbrinke, thanks for looking around to this. I've made a couple of attempts to fix or investigate this but I don't get any further. What I find very weird is I'm only seeing this behaviour when the string enum is a component/outer enum

This is currently existing in the petstore sample and the generated Elixir client in the repository.

The pet store 3_0 enum test:
https://github.com/OpenAPITools/openapi-generator/blob/8e9a17fe0252c1290f585d1fb680c4e64cb5495a/modules/openapi-generator/src/test/resources/3_0/petstore-with-fake-endpoints-models-for-testing.yaml#L1653-L1689

generates the following module:
https://github.com/OpenAPITools/openapi-generator/blob/master/samples/client/petstore/elixir/lib/openapi_petstore/model/enum_test.ex  

Which here we can see there is no issue with `enum_string` being just a type of `String.t`; however the outer enum is generated into a module as seen with the Ory example https://github.com/OpenAPITools/openapi-generator/blob/master/samples/client/petstore/elixir/lib/openapi_petstore/model/outer_enum.ex

**Comment by binajmen** [comment]
@barttenbrinke @tobbbles, apologies for the ping. Is this bug being tracked and worked on? I'm having issues verifying sessions in my Elixir project, which renders the Ory client unusable.

```elixir
 Frontend.to_session(Ory.Connection.new(), Cookie: cookies(conn))

 [error] ** (FunctionClauseError) no function clause matching in Ory.Deserializer.to_struct/2
 ```

**Comment by barttenbrinke** [comment]
@binajmen The core issue is that there is no simple test / example that reproduces the issue at the moment. All enum tests work in the OpenAPITools suite and the simple test @tobbbles made also succeeds. I am not working on it atm, if somebody creates a test / example that reproduces the error I would be able to fix it quite quickly I think.

**Comment by binajmen** [comment]
I would love to help, but I'm clueless about where to start. @tobbbles, is there a reason why your test succeeded with the OpenAPITools suite?

**Comment by tobbbles** [comment]
Hi folks, finally got some vacation time to circle back on this. Are we sure the example I posted succeeds? With a fresh checkout I have the following; however in https://github.com/OpenAPITools/openapi-generator/pull/19435 where I introduce this test case, it looks like the CI is not executing it.

Could someone check to reproduce as I have please?

```sh
$ cat >samples/client/petstore/elixir/test/issue.exs<<EOF

defmodule IssueTest do
  use ExUnit.Case, async: true

  alias OpenapiPetstore.Deserializer
  alias OpenapiPetstore.Model.EnumTest

  @valid_json """
  {
    "enum_string": "UPPER",
    "outerEnum": "placed"
  }
  """

  test "jason_decode/2 with valid JSON" do
    assert Deserializer.jason_decode(@valid_json, EnumTest) ==
             {:ok,
              %EnumTest{
                enum_string: "UPPER",
                outerEnum: "placed"
              }}
  end
end

EOF 

$ cd samples/client/petstore/elixir

$ mix deps.get
...

$ mix test test/issue.exs
Running ExUnit with seed: 24428, max_cases: 64



  1) test jason_decode/2 with valid JSON (IssueTest)
     test/issue.exs:14
     ** (FunctionClauseError) no function clause matching in OpenapiPetstore.Deserializer.to_struct/2

     The following arguments were given to OpenapiPetstore.Deserializer.to_struct/2:

         # 1
         "placed"

         # 2
         OpenapiPetstore.Model.OuterEnum

     Attempted function clauses (showing 3 out of 3):

         defp to_struct(nil, _)
         defp to_struct(list, module) when is_list(list) and is_atom(module)
         defp to_struct(map, module) when is_map(map) and is_atom(module)

     code: assert Deserializer.jason_decode(@valid_json, EnumTest) ==
     stacktrace:
       (openapi_petstore 1.0.0) lib/openapi_petstore/deserializer.ex:97: OpenapiPetstore.Deserializer.to_struct/2
       (elixir 1.17.2) lib/map.ex:916: Map.update!/3
       (openapi_petstore 1.0.0) lib/openapi_petstore/model/enum_test.ex:36: OpenapiPetstore.Model.EnumTest.decode/1
       (openapi_petstore 1.0.0) lib/openapi_petstore/deserializer.ex:19: OpenapiPetstore.Deserializer.jason_decode/2
       test/issue.exs:15: (test)


Finished in 0.05 seconds (0.05s async, 0.00s sync)
1 test, 1 failure
```

**Comment by tobbbles** [comment]
I think the main root issue here is the generated enum modules are just empty struct,

```exlixir
iex(3)> OpenapiPetstore.Model.OuterEnum.__struct__
%OpenapiPetstore.Model.OuterEnum{}
```

Which is more or less unusable.
